pkg/sql/logictest/testdata/logic_test/mixed_version_sequence_per_node_cache

```diff
@@ -0,0 +1,104 @@
+# LogicTest: cockroach-go-testserver-23.1
+
+# Verify that all nodes are running the previous version.
+
+query T nodeidx=0
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+query T nodeidx=1
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+query T nodeidx=2
+SELECT crdb_internal.node_executable_version()
+----
+23.1
+
+statement error pq: at or near "per": syntax error
+CREATE SEQUENCE cache_test PER NODE CACHE 10
+
+# Upgrade one node to 24.1
+
+upgrade 0
+
+# Verify that node index 0 is now running 24.1 binary.
+
+query T nodeidx=0
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+statement error pgcode 0A000 node-level cache not supported before V24.1
+CREATE SEQUENCE cache_test PER NODE CACHE 10
+
+upgrade 1
+
+upgrade 2
+
+query T nodeidx=1
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+query T nodeidx=2
+SELECT crdb_internal.release_series(crdb_internal.node_executable_version())
+----
+24.1
+
+query B retry
+SELECT crdb_internal.is_at_least_version('23.2-4')
```

Comment:
from Jasmine: the problem is that for the version gates i'm using clusterversion.V24_1 which is latest = 23.2-4

from me: oh shoot you're right - i think the only thing im "worried" about then is that when the latest internal version changes, we will have to update this test, but this might be common